### ISSUE TITLE
Errno:: EBADF on Open3.popen3

### ISSUE THREAD

**Issue by yamam** [jruby/jruby#5766]
### Environment
jruby 9.1.17.0 (2.3.3) 2019-02-26 fffffff OpenJDK 64-Bit Server VM 11.0.3+7-Ubuntu-1ubuntu218.04.1 on 11.0.3+7-Ubuntu-1ubuntu218.04.1 +jit [linux-x86_64]

### Expected Behavior

In CRuby the following command succeeds.

```
ruby -r open3 -e 'Open3.popen3("ls")'
```

### Actual Behavior

In jruby the following error occurs.

```
Errno::EBADF: Bad file descriptor - ls
      spawn at org/jruby/RubyProcess.java:1566
      spawn at org/jruby/RubyKernel.java:1535
  popen_run at /usr/share/jruby/lib/ruby/stdlib/open3.rb:206
     popen3 at /usr/share/jruby/lib/ruby/stdlib/open3.rb:102
     <main> at -e:1
```

**Comment by yamam** [comment]
This problem was fixed in 9.2.7.0.

**Comment by ggongaware** [comment]
This problem seems to have returned in 9.2.9.0
